pkg/kvstore/store/store.go

```diff
@@ -297,7 +297,7 @@ func (s *SharedStore) syncLocalKeys(ctx context.Context) error {
 	// Create a copy of all local keys so we can unlock and sync to kvstore
 	// without holding the lock
 	s.mutex.RLock()
-	keys := []LocalKey{}
+	keys := make([]LocalKey, len(s.localKeys))
 	for _, key := range s.localKeys {
 		keys = append(keys, key)
 	}
```

Comment:
This doesn't do what you want as you'll end up with a slice that is `2*len(s.localKeys)` large with the first half of values being the zero value of `LocalKey` after the `append` loop. What you want to do instead is to preallocate a slice of _capacity_ `len(s.localKeys)`, ie: `keys := make([]LocalKey, 0, len(s.localKeys))`.